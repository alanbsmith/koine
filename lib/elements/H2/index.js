import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';
import { rem } from 'polished';

import { fontWeights } from '../../modifiers';

const H2 = styled.h2`
  font-size: ${rem(36)};
  margin: 0;
  text-transform: capitalize;
  ${applyStyleModifiers(fontWeights)}
`;

export default H2;
