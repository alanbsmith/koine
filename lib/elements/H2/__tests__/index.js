import React from 'react';
import H2 from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('H2 Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <H2>Header Element</H2>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies mediumWeight modifier', () => {
    const tree = renderWithTheme(
      <H2 modifiers={['mediumWeight']}>Header Element</H2>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
