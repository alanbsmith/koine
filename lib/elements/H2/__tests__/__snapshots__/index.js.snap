// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A H2 Element applies mediumWeight modifier 1`] = `
<h2
  className="sc-bdVaJa cWLNgv"
>
  Header Element
</h2>
`;

exports[`A H2 Element renders correctly 1`] = `
<h2
  className="sc-bdVaJa cWLNgv"
>
  Header Element
</h2>
`;
